gradle-module-plugin/src/main/kotlin/io/ia/sdk/gradle/modl/extension/ModuleSettings.kt

```diff
@@ -83,6 +84,43 @@ open class ModuleSettings @javax.inject.Inject constructor(objects: ObjectFactor
      */
     val moduleDependencies: MapProperty<String, String> = objects.mapProperty(String::class.java, String::class.java)
 
+    /**
+     * New version of moduleDependencies for 8.3+ which allows for the addition of a "required" flag to be specified.
+     * Uses a builder to construct the property so the ModuleDependencySpec piece can be extrapolated away.
+     * Note: This required flag will only be put in the XML if requiredIgnitionVersion is explicitly set to 8.3+.
+     *
+     * ### Examples:
+     *
+     * _Groovy_
+     *   moduleDependencySpecs = [
+     *      moduleId("com.inductiveautomation.vision") {
+     *          it.scope = "GCD"
+     *          it.required = true
+     *      }
+     *   ]
+     *
+     * _Kotlin_
+     *   moduleDependencySpecs = setOf(
+     *      moduleId("com.inductiveautomation.vision") {
+     *          scope = "GCD"
+     *          required = true
+     *      }
+     *   )
+     */
+    val moduleDependencySpecs: SetProperty<ModuleDependencySpec> = objects.setProperty(
+        ModuleDependencySpec::class.java
+    ).convention(
```

Comment:
Yeah the convention worked for kotlin builds (it would default to getting its data from the legacy fields correctly and the additional conditional wasn't necessary in WriteModuleXML), but for some reason it didn't work with Groovy builds. Hence the additional conditional. I agree with taking it out in this case.